common/pose.h

```diff
@@ -12,48 +15,43 @@ namespace Pose {
 
 // A generic vector template
 template<class T>
-using Vector = T[3];
-
-/*
- * A class for objects which can return their position in space.
- */
-template<class LengthUnit>
-class HasPosition
-{
-public:
-    virtual Vector<LengthUnit> &getPosition() = 0;
-}; // HasPosition
-
-/* 
- * A class for objects which can return their attitude (rotation).
- */
-template<class AngleUnit>
-class HasAttitude
-{
-public:
-    virtual Vector<AngleUnit> &getAttitude() = 0;
-};
+using Triple = std::tuple<T, T, T>;
```

Comment:
Tuples **are** cool but, because they're immutable, they are also a pain in the ass - why not just use the array-based one throughout?

Reply:
Tuples are a pain in the ass, but the problem with using arrays is that we want templatable getters (so we can choose to get position as e.g. metres) and if we can't return an array declared on the stack. I suppose we could do:
``
template<class LengthUnit>
void getPosition(Array3<LengthUnit> &position);
``
But that would mean that users of the function would have to declare their own arrays before calling ``getPosition``.

Another possibility (for the future) is that we use [Eigen](http://eigen.tuxfamily.org)'s template vector classes (though I'm not certain that the templating in the units lib will necessarily place nicely with Eigen).

Reply:
How about ``std::array<LengthUnit>``?

Reply:
e.g. something like this (pretty sure you can instantiate an array like this):

```c++
template<class OutputUnit, class InputType>
inline std::array<OutputUnit, 3> makeUnitTriple(const std::array<InputType, 3> &values)
{
    return {static_cast<OutputUnit>(values[0]), 
                static_cast<OutputUnit>(values[1]), 
                static_cast<OutputUnit>(values[2])};
}
```

Reply:
Hmm this seems to work. It just seems not to work if you use C-style arrays :-/.

Ok well this is def preferable!

Reply:
Not being able to return C-style arrays does seem really stupid - if you want to fill your stack with array by returning them from functions why shouldn't you be able to!